lib/cylc/remote.py

```diff
@@ -24,16 +24,21 @@
 from time import sleep
 from pipes import quote
 from posix import WIFSIGNALED
+from random import shuffle, choice
+import unittest
```

Comment:
Hi @sadielbartholomew ! I'm lurking here, receiving notifications for updates - not that I understand everything here - but also planning to test this pull request later, as I was playing with SSH and cylc anyway.

While quickly going through the changes today, just to have an idea of what the change will look like, I noticed the `unittest` module in this file. Wouldn't it be possible to move the `unittest` import and related classes/calls, to under the `tests/` folder?

By the way, I am reading your comments/docstrings, and they are super clear and useful! Thanks for that!

Reply:
@kinow - shouldn't you be asleep :grin: I must say I just assumed unittests should be in the source file being tested, but maybe you know better, e.g.: https://stackoverflow.com/a/1341053

Reply:
> shouldn't you be asleep

Both of you!

Reply:
@hjoliver was going to, but then had one of these [epiphanies](https://i.redd.it/mdjsol1eiad11.jpg) and one issue takes to another pull request, that takes to another repository... Oh, if other files include both code + tests, then all good for me. I'm used with the tests being located separately from the rest of the code, but if the current tests work, coverage is generated OK, etc, so I wouldn't like to distract @sadielbartholomew from the main work in this issue (apologies if I already did so!) :-)